sample.html

```diff
@@ -2021,8 +2021,9 @@ <h4 class="panel-title">Options</h4>
             type: $("input[name='serialType']:checked").val(),
             data: $("#serialCmd").val()
         };
-        //allow some escape characters
-        serialData.data = serialData.data.replace(/\\n/g, "\n").replace(/\\r/g, "\r").replace(/\\t/g, "\t");
+        //allow some escape characters (newlines, tabs, hex/unicode)
+        serialData.data = serialData.data.replace(/\\n/g, "\n").replace(/\\r/g, "\r").replace(/\\t/g, "\t")
+            .replace(/\\[xu]([\dA-Fa-f]+)/g, (m, s1) => String.fromCharCode(parseInt(s1, 16)));
```

Comment:
This arrow function breaks IE11 support

Reply:
Can this be rewritten like so?
```js
        serialData.data = serialData.data.replace(/\\n/g, "\n").replace(/\\r/g, "\r").replace(/\\t/g, "\t")
            .replace(/\\[xu]([\dA-Fa-f]+)/g, (m, s1).then(function() {
                String.fromCharCode(parseInt(s1, 16))
            })
        );
```

Reply:
yes, but i found a case in retesting where the \x codes can try to collect too many values to convert, so I pushed up a change where the function is a variable to be used for both